### PR TITLE
Force annotations to newlines on properties

### PR DESCRIPTION
**Pull request by yissachar**
There are several issues dealing with the surprising fact that annotations end up on the same line as the target (https://github.com/facebook/ktfmt/issues/226, https://github.com/facebook/ktfmt/issues/395, https://github.com/facebook/ktfmt/issues/409). 

@strulovich From what I understand of your comments on these issues, it is uncontroversial to at least have multi-line annotations for properties. This PR attempts that.

### CODE REVIEW

File: core/src/test/java/com/facebook/ktfmt/format/FormatterTest.kt [hunk 1]
```diff
@@ -2370,6 +2375,24 @@ class FormatterTest {
               .trimMargin(),
           deduceMaxWidth = true)
 
+
+  @Test
+  fun `a constructor with many arguments over multiple lines and annotations`() =
```
Comment (yissachar) [inline core/src/test/java/com/facebook/ktfmt/format/FormatterTest.kt:2380]: This is the existing behaviour and unchanged by this PR but I didn't see any test that covered this.

File: core/src/main/java/com/facebook/ktfmt/format/KotlinInputAstVisitor.kt [hunk 2]
```diff
@@ -1854,6 +1855,16 @@ class KotlinInputAstVisitor(
         null, // Type-arguments are included in the annotation's callee expression.
         annotationEntry.valueArgumentList,
         listOf())
+
+    // Annotations on properties should each be on their own line
+    if (
+      annotationEntry.parent != null &&
+      annotationEntry.parent::class == KtDeclarationModifierList::class &&
+      annotationEntry.parent.parent != null &&
+      annotationEntry.parent.parent::class == KtProperty::class
```
Comment (yissachar) [inline core/src/main/java/com/facebook/ktfmt/format/KotlinInputAstVisitor.kt:1864]: I'm not sure that this check needs to be so verbose and guarded. Possibly `annotationEntry.parent.parent::class == KtProperty::class` is sufficient?

**Comment by ForsHant** [comment]
This is a very important one. Hope this will be released ASAP

**Comment by yissachar** [comment]
@davidtorosyan Thoughts on this?

**Comment by davidtorosyan** [comment]
@yissachar mind signing the CLA (and maybe rebasing)? I'd like to import your changes so we review them internally.

**Comment by yissachar** [comment]
Ah sorry, we've signed the CLA but I need to be added to the membership list. Let me sort this out.

**Comment by yissachar** [comment]
@davidtorosyan I sorted out the CLA now and merged main

**Comment by davidtorosyan** [comment]
Thanks @yissachar! We're taking a look now.

**Comment by yissachar** [comment]
Hi @davidtorosyan, any movement on this?

**Comment by davidtorosyan** [comment]
Hey @yissachar, apologies for the late reply, was travelling this week.

Sorry, but we aren't accepting formatting changes that are contested (i.e. not unanimously accepted when we ask a few people). We do this to avoid formatting arguments, and minimize churn on the project and its users.

This policy isn't one we've codified as of yet, but probably need to so we can make sure community effort is directed in the right direction (mostly fixing bugs and handling new features). With that said, thank you again for your effort. It does mean a lot to us.

**Comment by yissachar** [comment]
No worries, I understand the decision. Thanks for taking a look!

**Pull request closed without merging**